src/Main.java

```diff
@@ -0,0 +1,363 @@
+        }
+        /*System.out.println();
+        for (int i = 0; i < givenArray.length; i++) {
+            for (int j = 0; j < givenArray[i].length; j++) {
+                System.out.print(givenArray[i][j] + "\t");
+            }
+            System.out.println();
+        }*/
+    }
+
+    public static void swapOddAndEvenLinesIn2DArray(int[][] givenArray) {
+        printTwoDimensionalArray(givenArray);
+        for (int i = 0; i < givenArray.length; i++) {
+            int[] tmp = new int[givenArray.length];
+            if (i % 2 != 0) {
+                tmp = givenArray[i-1];
+                givenArray[i-1] = givenArray[i];
+                givenArray[i] = tmp;
+            }
+        }
+        System.out.println();
+        printTwoDimensionalArray(givenArray);
+    }
+
+    public static int[] sumOfLine(int[][] givenArray) {
+        printTwoDimensionalArray(givenArray);
+        System.out.println();
+        int[] array = new int[givenArray.length];
+        for (int i = 0; i < givenArray.length; i++) {
+            int sumOfLine = 0;
+            for (int j = 0; j < givenArray[i].length; j++) {
+                sumOfLine += givenArray[i][j];
+            }
+            array[i] = sumOfLine;
+        }
+        return array;
+    }
+
+    public static int findMinElement(int[][] givenArray) {
+        int minElement = givenArray[0][0];
+        for (int i = 0; i < givenArray.length; i++) {
+            for (int j = 0; j < givenArray[i].length; j++) {
+                if (givenArray[i][j] < minElement) {
+                    minElement = givenArray[i][j];
+                }
+            }
+        }
+        return minElement;
+    }
+
+    public static int findMaxElement(int[][] givenArray) {
+        int maxElement = givenArray[0][0];
+        for (int i = 0; i < givenArray.length; i++) {
+            for (int j = 0; j < givenArray[i].length; j++) {
+                if (givenArray[i][j] > maxElement) {
+                    maxElement = givenArray[i][j];
+                }
+            }
+        }
+        return maxElement;
+    }
+
+    public static int[][] swapMinAndMaxElementsInTwoDimensionalArray (int[][] givenArray) {
+        int minElement = findMinElement(givenArray);
+        int maxElement = findMaxElement(givenArray);
+        for (int i = 0; i < givenArray.length; i++) {
+            for (int j = 0; j < givenArray[i].length; j++) {
+                if (givenArray[i][j] == minElement) {
+                    givenArray[i][j] = maxElement;
+                }
+                else if (givenArray[i][j] == maxElement) {
+                    givenArray[i][j] = minElement;
+                }
+            }
+        }
+        return givenArray;
+    }
+
+    public static int[][] swapMinAndMaxElementsInTwoDimensionalArray2 (int[][] givenArray) {
+        int minElement = givenArray[0][0];
+        int maxElement = givenArray[0][0];
+        for (int i = 0; i < givenArray.length; i++) {
```

Comment:
do you really need those 4 for loops? why are you traversing array twice?

Reply:
Yes I need it. At first for cycle I search min and max element. I can\t find elements before cycle finish. And at second for I search already defined min and max element and swap them.